azext_edge/edge/providers/support/base.py

```diff
@@ -180,164 +180,83 @@ def process_deployments(
-        # TODO: Workaround
-        r.api_version = replicasets.api_version
-        r.kind = "Replicaset"
-        replicaset_metadata: V1ObjectMeta = r.metadata
-        replicaset_namespace: str = replicaset_metadata.namespace
-        replicaset_name: str = replicaset_metadata.name
-
-        if prefix_names:
-            matched_prefix = [replicaset_name.startswith(prefix) for prefix in prefix_names]
-            if not any(matched_prefix):
-                continue
-
-        processed.append(
-            {
-                "data": generic.sanitize_for_serialization(obj=r),
-                "zinfo": f"{replicaset_namespace}/{resource_api.moniker}/replicaset.{replicaset_name}.yaml",
-            }
-        )
-
-    return processed
+    return _process_kubernetes_resources(
+        child_group=child_group,
+        resources=v1_apps.list_replica_set_for_all_namespaces(label_selector=label_selector),
+        resource_api=resource_api,
+        prefix_names=prefix_names,
+        kind="Replicaset",
+    )
 
 
 def process_daemonsets(
     resource_api: EdgeResourceApi,
-    label_selector: str = None,
-    field_selector: str = None,
-    prefix_names: List[str] = None,
-):
-    from kubernetes.client.models import V1DaemonSet, V1DaemonSetList
-
+    child_group: str = "",
+    field_selector: Optional[str] = None,
+    label_selector: Optional[str] = None,
+    prefix_names: Optional[List[str]] = None,
+) -> List[dict]:
     v1_apps = client.AppsV1Api()
 
-    processed = []
-    if not prefix_names:
-        prefix_names = []
-
-    daemonsets: V1DaemonSetList = v1_apps.list_daemon_set_for_all_namespaces(
-        label_selector=label_selector, field_selector=field_selector
+    return _process_kubernetes_resources(
+        child_group=child_group,
+        resources=v1_apps.list_daemon_set_for_all_namespaces(
+            label_selector=label_selector,
+            field_selector=field_selector
+        ),
+        resource_api=resource_api,
+        prefix_names=prefix_names,
+        kind="Daemonset",
     )
-    logger.info(f"Detected {len(daemonsets.items)} daemonsets.")
-
-    for daemonset in daemonsets.items:
-        d: V1DaemonSet = daemonset
-        d.api_version = daemonsets.api_version
-        d.kind = "Daemonset"
-        daemonset_metadata: V1ObjectMeta = d.metadata
-        daemonset_namespace: str = daemonset_metadata.namespace
-        daemonset_name: str = daemonset_metadata.name
-
-        if prefix_names:
-            matched_prefix = [daemonset_name.startswith(prefix) for prefix in prefix_names]
-            if not any(matched_prefix):
-                continue
-
-        processed.append(
-            {
-                "data": generic.sanitize_for_serialization(obj=d),
-                "zinfo": f"{daemonset_namespace}/{resource_api.moniker}/daemonset.{daemonset_name}.yaml",
-            }
-        )
-
-    return processed
 
 
-def process_nodes():
+def process_nodes() -> dict:
```

Comment:
for these type hints, can you be more specific as to what kind of dict? Dict[str, str]? 
if it is something more complex than 2 or 3 layers (Dict[str, Union[str, Dict[str,str]]) then you can leave it as dict

Reply:
the key is str and map can be str or v1node, so I will state as Dict[str, object]?